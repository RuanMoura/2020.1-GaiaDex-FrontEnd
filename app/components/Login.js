import { StatusBar } from 'expo-status-bar';
import React from 'react';
import Svg, { Path } from "react-native-svg";
import { Platform, 
    StyleSheet, 
    Text,
    Button, 
    View, 
    StatusBar as sBar, 
    TextInput, 
    TouchableOpacity
} from 'react-native';

export default function Regform() {
  return (
    <View style={styles.Regform}>

      <Text style={styles.header}>Seja bem-vindo</Text>

      <TextInput style={styles.textinput} placeholder='Seu e-mail:'
      underlineColorAndroid={'transparent'}/>

      <TextInput style={styles.textinput} placeholder='Sua senha:'
      securyTextEntry={true} underlineColorAndroid={'transparent'}/>

      <Text style={styles.loginText}>Login</Text>

      <TouchableOpacity style={styles.button}>
        <svg width="24" height="15" viewBox="0 0 24 15" fill="none" xmlns="http://www.w3.org/2000/svg">
        <path d="M23.0993 6.93273L15.8993 0.171614C15.7118 -0.00447041 15.4081 -0.00447041 15.2206 0.171614C15.0331 0.347699 15.0331 0.632934 15.2206 0.808976L21.6012 6.80068H0.680026C0.414707 6.80068 0.200012 7.00228 0.200012 7.25143C0.200012 7.50058 0.414707 7.70219 0.680026 7.70219H21.6012L15.2206 13.6938C15.0331 13.8699 15.0331 14.1552 15.2206 14.3312C15.3143 14.4192 15.4372 14.4633 15.56 14.4633C15.6828 14.4633 15.8056 14.4192 15.8994 14.3312L23.0994 7.57009C23.2868 7.39405 23.2868 7.10881 23.0993 6.93273Z" fill="#F2E0F5"/>
        </svg>
      </TouchableOpacity>

      <TouchableOpacity style={styles.button_Fb_Google}>
        <Text>Fazer login com Facebook</Text>
        <Svg width={19} height={19} viewBox="0 0 19 19" fill="none">
          <Path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M2.883 0A2.883 2.883 0 000 2.883v13.234A2.883 2.883 0 002.883 19h13.234A2.883 2.883 0 0019 16.117V2.883A2.883 2.883 0 0016.117 0H2.883zm4.61 18.877V12.44h-2.97V9.41h2.97V7.102c0-2.626 1.745-4.077 4.416-4.077 1.28 0 2.617.205 2.617.205v2.578h-1.474c-1.452 0-1.905.808-1.905 1.636v1.967h3.242l-.518 3.03h-2.724v6.436H7.493z"
            fill="#F2E0F5"
          />
        </Svg>
      </TouchableOpacity>

      <TouchableOpacity style={styles.button_Fb_Google}>
        <Text>Fazer login com Google</Text>
        <Svg width={21} height={20} viewBox="0 0 21 20" fill="none">
          <Path
          d="M20.88 10.168c0 5.658-4.003 9.684-9.915 9.684C5.297 19.852.72 15.422.72 9.936.72 4.45 5.297.02 10.965.02c2.76 0 5.082.98 6.87 2.595L15.047 5.21C11.399 1.803 4.616 4.362 4.616 9.936c0 3.458 2.854 6.261 6.35 6.261 4.056 0 5.576-2.814 5.816-4.274h-5.817v-3.41h9.754c.095.507.161.995.161 1.655z"
          fill="#F2E0F5"
          />
        </Svg>
      </TouchableOpacity>

      <View>
        <Button style={styles.text_Register_Forget_Password} title="Cadastro" />
        <Button style={styles.text_Register_Forget_Password} title="Esqueci a senha" />
      </View>

      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  Regform: {
    alignSelf: 'stretch',
  },
  header: {
    fontFamily: 'Inter',
    fontStyle: 'normal',
    fontWeight: 600,
    fontSize: '2.76em',
    lineHeight: '3.3em',
    color: '#F2E0F5',
  },
  textinput: {
    height: '8%',
    width: '83%',
    marginLeft: '8.5%',
    marginRight: '8.5%',
    background: 'rgba(36, 37, 40, 0.1)',
    borderRadius: '20px',
    border: 'transparent',
    fontFamily: 'Inter',
    fontStyle: 'normal',
    fontWeight: 500,
    fontSize: '1.08em',
    lineHeight: '100%',
    color: '#3C3C3C',
  },
  loginText: {
    fontFamily: 'Inter',
    fontStyle: 'normal',
    fontWeight: 600,
    fontSize: '1.92em',
    lineHeight: '2.3em',
    color: '#242528',
  },
  button: {
    borderRadius: '100%',
    width: '8.5vw',
    height: '8.5vw',
    background: '#4D525A',
  },
  button_Fb_Google: {
    height: '6.20%',
    width: '83%',
    marginLeft: '8.5%',
    marginRight: '8.5%',
    background: '#38143E',
    border: '0.95px solid #E15A97',
    boxSizing: 'border-box',
    borderRadius: '10px',
  },
  text_Fb_Google: {
    fontFamily: 'Inter',
    fontStyle: 'normal',
    fontWeight: 500,
    fontSize: '14px',
    lineHeight: '171%',
    textAlign: 'center',
    letterSpacing: '0.1px',
    color: '#F2E0F5',
  },
  text_Register_Forget_Password: {
    fontFamily: 'Inter',
    fontStyle: 'normal',
    fontWeight: '600',
    fontSize: '1.08em',
    lineHeight: '100%',
    display: 'flex',
    alignItems: 'center',
    textAlign: 'center',
    color: '#242528',
    boxShadow: 'inset 0 -0.5vw 0 white, inset 0 -1.2vw 0 #19BB53', /* #E15A97 */
  },
});
