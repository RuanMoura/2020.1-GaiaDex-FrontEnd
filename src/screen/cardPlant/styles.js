import { StyleSheet, Dimensions } from 'react-native';
import { gray, green, purple } from '../../theme/colorPalette';

const { height, width } = Dimensions.get('screen');

const styles = StyleSheet.create({
  containerView: {
    flexDirection: 'column',
    backgroundColor: gray.iron(),
    width,
    // backgroundColor: gray.shark(),
    flex: 1,
  },
  containerHeader: {
    flex: 0.1,
    backgroundColor: gray.shark(),
    width,
    alignSelf: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  headerIcons: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  containerPlant: {
    flex: 0.3,
    backgroundColor: gray.shark(),
    flexDirection: 'column',
  },
  plantImg: {
    flex: 0.4,
    width: width / 2.1,
    alignSelf: 'center',
    marginTop: '-9%',
  },
  plantInfo: {
    backgroundColor: gray.shark(),
    width: width / 1.3,
    flex: 0.25,
    flexDirection: 'column',
    alignItems: 'flex-start',
    alignSelf: 'center',
  },
  plantInfoTitle: {
    flex: 0.6,
    color: gray.iron(),
    fontSize: 23,
  },
  plantInfoStats: {
    flex: 0.4,
    flexDirection: 'row',
    width: width / 1.3,
    justifyContent: 'space-between',
    alignContent: 'center',
    alignItems: 'center',
  },
  plantInfoDiv: {
    width: '35%',
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  plantInfoText: {
    color: gray.iron(),
    fontSize: 10,
  },
  plantText: {
    backgroundColor: gray.shark(),
    flex: 0.375,
    width: width / 1.3,
    alignSelf: 'center',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  plantTextDescription: {
    color: gray.iron(),
    textAlign: 'left',
    padding: '4%',
    fontSize: 10,
  },
  containerBody: {
    flex: 0.6,
    backgroundColor: purple.frenchLilacLighter(),
    borderTopRightRadius: 20,
    borderTopLeftRadius: 20,
  },
  menuBar: {
    flexDirection: 'row',
    alignItems: 'center',
    alignSelf: 'center',
    justifyContent: 'space-around',
    width: width / 1.7,
  },
  menuBarTab: {
    width: width / 4,
    alignItems: 'center',
    justifyContent: 'center',
  },
  menuBarTabText: {
    color: gray.iron(),
    fontSize: 12,
  },
  bodyPlants: {
    width: width / 1.2,
    alignSelf: 'center',
    padding: 10,
    flex: 0.35,
  },
  bodyPlantsTitle: {
    color: gray.shark(),
    fontSize: 16,
    marginBottom: 10,
  },
  userDiv: {
    padding: 5,
    width: width / 4.5,
    alignItems: 'center',
  },
  UserImg: {
    width: width / 5,
    height: width / 5,
  },
  userName: {
    fontSize: 12,
    textAlign: 'center',
  },
  bodyTopics: {
    width: width / 1.2,
    alignSelf: 'center',
    padding: 0,
    flex: 0.65,
  },
  TopicDivContainer: {
    width: width / 1.2,
    alignSelf: 'center',
    marginVertical: '3%',
    height: height / 3.8,
  },
  TopicImg: {
    width: '100%',
    height: height / 3.8,
    alignSelf: 'center',
    flexDirection: 'column-reverse',
    alignItems: 'center',
  },
  TopicDescriptionDiv: {
    width: '100%',
    flex: 0.42,
    backgroundColor: green.darkFern(),
    opacity: 0.8,
    flexDirection: 'column',
    justifyContent: 'space-around',
    paddingLeft: 15,
  },
  TopicCommentsDiv: {
    width: '100%',
    flex: 0.18,
    backgroundColor: green.zuccini(),
    borderBottomRightRadius: 20,
    borderBottomLeftRadius: 20,
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 15,
  },
  TopicUsername: {
    color: gray.iron(),
    fontWeight: '400',
    fontSize: 13,
  },
  TopicTitle: {
    color: gray.iron(),
    fontWeight: 'bold',
    fontSize: 13,
  },
  TopicDescription: {
    color: gray.iron(),
    fontWeight: '500',
    fontSize: 10,
  },

  ButtonBackground: {
    position: 'absolute',
    zIndex: 0,
    marginTop: height / 1.2,
    marginLeft: width / 1.35,
    width: width / 5,
    height: width / 5,
    borderRadius: 100,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: purple.lightWisteria(0.3),
  },
  ButtonDiv: {
    width: width / 6.8,
    height: width / 6.8,
    borderRadius: 100,
    backgroundColor: gray.shark(),
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default styles;
