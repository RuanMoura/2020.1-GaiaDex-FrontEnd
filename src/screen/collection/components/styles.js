import { StyleSheet, Dimensions } from 'react-native';

const { height, width } = Dimensions.get('screen');
const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#F2E0F5',
  },
  header: {
    flex: 0.25,
    flexDirection: 'column',
    width: width / 1.1,
    alignSelf: 'center',
  },
  headerIcon: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'flex-start',
    flex: 0.6,
  },
  headerTitle: {
    width: width / 1.2,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  headerTitleText: {
    fontSize: 35,
    fontWeight: '600',
    paddingLeft: 5,
  },
  body: {
    flex: 0.7,
    alignItems: 'center',
    paddingBottom: 20,
  },
  plant: {
    flex: 0.45,
    maxHeight: height / 3,
    width: width / 1.2,
    flexDirection: 'row',
    marginBottom: '10%',
  },
  plantBackground: {
    flex: 1,
  },
  plantDescription: {
    flex: 1,
    backgroundColor: '#59A44D',
    fontWeight: '600',
    height: '50%',
    alignSelf: 'center',
    marginTop: '10%',
    borderTopRightRadius: 20,
    borderBottomRightRadius: 20,
    flexDirection: 'column',
    alignItems: 'center',
    padding: 10,
  },
  plantDescriptionTitle: {
    fontSize: 14,
    color: 'white',
    paddingBottom: 5,
    textAlign: 'center',
  },
  plantDescriptionText: {
    fontSize: 12,
    color: 'white',
    textAlign: 'center',
  },
  plantButtonDiv: {
    flexDirection: 'row',
    width: '100%',
    height: '50%',
    alignSelf: 'flex-start',
    margin: 5,
    marginTop: '4%',
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  plantButton: {
    width: '45%',
    height: '80%',
    backgroundColor: 'orange',
    alignSelf: 'flex-start',
    margin: 5,
    marginTop: '5%',
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
  deleteButton: {
    width: '45%',
    height: '80%',
    backgroundColor: 'red',
    alignSelf: 'flex-start',
    margin: 5,
    marginTop: '5%',
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
  deleteButtonText: {
    fontSize: 12,
    color: 'white',
  },
});
export default styles;
