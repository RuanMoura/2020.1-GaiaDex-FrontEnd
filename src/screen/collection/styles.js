import { StyleSheet, Dimensions } from 'react-native';
import { gray, highlight, purple } from '../../theme/colorPalette';

const { height, width } = Dimensions.get('screen');

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: purple.frenchLilacLighter(),
    // paddingTop: Constants.statusBarHeight,
  },
  editView: {
    backgroundColor: purple.frenchLilacLighter(),
    width: (width * 4) / 5,
    height: (height * 2) / 9,
    marginTop: (height * 5) / 36,
    marginLeft: width / 10,
    borderRadius: 10,
    shadowColor: gray.darkest(),
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 50,
    paddingHorizontal: width / 25,
    paddingVertical: height / 90,
    justifyContent: 'space-between',
  },
  editTitle: {
    fontSize: 17,
  },
  editTextInput: {
    height: 40,
    width: (width * 4) / 5 - (width * 2) / 25,
    borderBottomColor: purple.grapeDark(),
    borderBottomWidth: 1,
  },
  editButton: {
    fontSize: 14,
    color: highlight.lochmara(),
    marginLeft: (width * 9) / 125,
  },
  exploreContainer: {
    flexDirection: 'row',
    alignSelf: 'center',
    width: width / 1.1,
    flex: 0.22,
    justifyContent: 'space-between',
    alignContent: 'center',
    alignItems: 'center',
  },
  exploreText: {
    fontSize: 40,
    fontWeight: '600',
    color: gray.outerSpace(),
  },
  popularContainer: {
    flexDirection: 'column',
    alignSelf: 'center',
    width: width / 1.1,
    flex: 0.7,
    justifyContent: 'flex-start',
  },
  tabContainer: {
    flexDirection: 'row',
    width: width / 1.7,
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: '10%',
  },
  tabText: {
    fontSize: 18,
    fontWeight: '400',
  },
  tabTextactivate: {
    fontSize: 18,
    fontWeight: '500',
    textDecorationLine: 'underline',
    textDecorationStyle: 'solid',
    textDecorationColor: gray.darkest(),
  },
  popularTabs: {
    flexDirection: 'row',
    flex: 0.2,
  },
  popularTab: {
    fontSize: 22,
    fontWeight: '400',
    textDecorationLine: 'underline',
    textDecorationStyle: 'solid',
    textDecorationColor: gray.darkest(),
  },
  popularContent: {
    flex: 0.8,
  },
  FavoritePlant: {
    width: width / 3,
    flex: 1,
    flexDirection: 'column-reverse',
    borderRadius: 30,
    marginRight: 15,
    backgroundColor: purple.grapeDark(0.9),
  },
  FavoritePlantImg: {
    width: '100%',
    resizeMode: 'cover',
    flexDirection: 'column-reverse',
    flex: 1,
    justifyContent: 'space-between',
  },
  plantInfo: {
    backgroundColor: purple.grapeDark(0.9),
    flex: 0.3,
    borderBottomRightRadius: 30,
    borderBottomLeftRadius: 30,
    width: width / 3,
    alignSelf: 'center',
  },
  myPlantsContainer: {
    flexDirection: 'column',
    alignSelf: 'center',
    width: width / 1.1,
    flex: 1,
    justifyContent: 'flex-start',
  },
  myPlantsText: {
    paddingTop: 30,
    color: gray.outerSpace(),
    fontSize: 26,
    fontWeight: '600',
  },
  myplantPlant: {
    marginTop: 10,
    width: width / 2.3,
    height: height / 4,
    flexDirection: 'column-reverse',
    borderRadius: 20,
    marginRight: 15,
    backgroundColor: 'brown',
  },
  myplantInfo: {
    backgroundColor: purple.grapeDark(0.9),
    flex: 0.3,
    borderBottomRightRadius: 20,
    borderBottomLeftRadius: 20,
    width: width / 2.3,
    alignItems: 'center',
    justifyContent: 'space-around',
    flexDirection: 'row',
  },
  plantText: {
    padding: 5,
    fontSize: 10,
    textAlign: 'center',
    color: purple.frenchLilacLighter(),
  },
});
export default styles;
