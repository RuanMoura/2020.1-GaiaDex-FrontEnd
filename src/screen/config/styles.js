import { StyleSheet, Dimensions } from 'react-native';

const largura = Dimensions.get('screen').width;
const altura = Dimensions.get('screen').height;

const styles = StyleSheet.create({
  containerView: {
    flex: 1,
    backgroundColor: 'white',
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
  containerHeader: {
    flex: 0.2,
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: largura,
    paddingRight: '8%',
  },
  containerHeaderIcon: {
    flexDirection: 'row',
  },
  headerTitle: {
    fontSize: 32,
    color: '#242528',
  },
  notificationContainer: {
    flex: 0.7,
  },
  notification: {
    flex: 0.1,
    flexDirection: 'row',
    marginVertical: '3%',
    justifyContent: 'space-between',
    alignItems: 'center',
    width: largura / 1.2,
  },
  notificationText: {
    fontSize: 15,
    fontWeight: '500',
    color: '#242528',
  },
});

export default styles;
