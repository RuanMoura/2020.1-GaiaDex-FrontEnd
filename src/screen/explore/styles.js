import { StyleSheet, Dimensions } from 'react-native';

import Constants from 'expo-constants';

const largura = Dimensions.get('screen').width;
const altura = Dimensions.get('screen').height;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#F2E0F5',
    paddingTop: Constants.statusBarHeight,
  },
  exploreContainer: {
    flexDirection: 'row',
    alignSelf: 'center',
    width: largura / 1.1,
    flex: 0.22,
    justifyContent: 'space-between',
    alignContent: 'center',
    alignItems: 'center',
  },
  exploreText: {
    fontSize: 40,
    fontWeight: '600',
    color: '#31353A',
  },
  popularContainer: {
    flexDirection: 'column',
    alignSelf: 'center',
    width: largura / 1.1,
    flex: 0.35,
    justifyContent: 'flex-start',
  },
  popularTabs: {
    flexDirection: 'row',
    flex: 0.2,
  },
  popularTab: {
    fontSize: 22,
    fontWeight: '400',
    textDecorationLine: 'underline',
    textDecorationStyle: 'solid',
    textDecorationColor: '#000',
  },
  popularContent: {
    flex: 0.8,
  },
  FavoritePlant: {
    width: largura / 2.7,
    height: altura / 4,
    flexDirection: 'column-reverse',
    borderRadius: 30,
    marginRight: 15,
    backgroundColor: 'rgba(52, 9, 37, 0.91)',
  },
  FavoritePlantImg: {
    width: '100%',
    resizeMode: 'cover',
    flexDirection: 'column-reverse',
    flex: 1,
  },
  plantInfo: {
    backgroundColor: 'rgba(52, 9, 37, 0.91)',
    flex: 0.2,
    borderBottomRightRadius: 20,
    borderBottomLeftRadius: 20,
    width: largura / 2.7,
    alignSelf: 'center',
  },
  myPlantsContainer: {
    flexDirection: 'column',
    alignSelf: 'center',
    width: largura / 1.1,
    flex: 0.3,
    justifyContent: 'flex-start',
  },
  myPlantsText: {
    paddingTop: 30,
    color: '#31353A',
    fontSize: 26,
    fontWeight: '600',
  },
  myplantPlant: {
    marginTop: 10,
    height: altura / 6,
    width: largura / 4.5,
    flexDirection: 'column-reverse',
    borderRadius: 20,
    marginRight: 15,
    backgroundColor: 'brown',
  },
  myplantInfo: {
    backgroundColor: 'rgba(52, 9, 37, 0.91)',
    flex: 0.3,
    borderBottomRightRadius: 10,
    borderBottomLeftRadius: 10,
    width: largura / 4.5,
    alignSelf: 'center',
  },
  plantText: {
    padding: 5,
    textAlign: 'center',
    fontSize: 10,
    color: 'white',
  },
});
export default styles;
