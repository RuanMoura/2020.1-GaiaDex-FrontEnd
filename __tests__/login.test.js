import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';

import renderer from 'react-test-renderer';

import { gray, purple } from '../src/theme/colorPalette';


import React from 'react';
import Login from '../src/screen/login/login';

import styles from '../src/screen/login/style';

jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);

let wrapper;

describe('<Login />', () => {
  beforeEach(() => {
    wrapper = mount(<Login />);
  });

  it('renders everything', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(wrapper.find('TextInput').length).toBe(2);
    expect(wrapper.find('View').length).toBe(26);
    expect(wrapper.find('Text').length).toBe(18);
    expect(wrapper.find('TouchableOpacity').length).toBe(5);
    expect(wrapper.find('StatusBar').length).toBe(1);
  });

  it('renders Text Input correctly', () => {
    expect(wrapper.find('TextInput').at(0)).toHaveStyle(styles.textinputLogin);
    expect(wrapper.find('TextInput').at(1)).toHaveStyle(styles.textinputLogin);
    expect(wrapper.find('TextInput').at(0)).toHaveProp(
      'placeholder',
      'Seu e-mail:'
    );
    expect(wrapper.find('TextInput').at(1)).toHaveProp(
      'placeholder',
      'Sua senha:'
    );
    expect(wrapper.find('TextInput').at(0)).toHaveProp(
      'underlineColorAndroid',
      'transparent'
    );
    expect(wrapper.find('TextInput').at(1)).toHaveProp(
      'underlineColorAndroid',
      'transparent'
    );
    expect(wrapper.find('TextInput').at(0)).toHaveProp('onChangeText');
    expect(wrapper.find('TextInput').at(1)).toHaveProp('onChangeText');
  });

  it('renders View correctly', () => {
    expect(wrapper.find('View').at(0)).toHaveStyle(styles.containerLogin);
    expect(wrapper.find('View').at(2)).toHaveStyle(styles.circle3);
    expect(wrapper.find('View').at(4)).toHaveStyle(styles.circle2);
    expect(wrapper.find('View').at(6)).toHaveStyle(styles.circle1);
    expect(wrapper.find('View').at(8)).toHaveStyle(styles.direction);
    expect(wrapper.find('View').at(14)).toHaveStyle(styles.viewbtnFbGoogle);
    expect(wrapper.find('View').at(20)).toHaveStyle(styles.direction);
  });
  it('renders Text correctly', () => {
    expect(wrapper.find('Text').at(0)).toHaveStyle(styles.headerLogin);
    expect(wrapper.find('Text').at(0)).toHaveText('Seja bem-vindo');
    expect(wrapper.find('Text').at(2)).toHaveStyle(styles.midTxt);
    expect(wrapper.find('Text').at(2)).toHaveText('Login');
    expect(wrapper.find('Text').at(8)).toHaveStyle(styles.txtBtnFbGoogleLogin);
    expect(wrapper.find('Text').at(8)).toHaveText('Fazer login com o Facebook');
    expect(wrapper.find('Text').at(12)).toHaveStyle(styles.txtBtnFbGoogleLogin);
    expect(wrapper.find('Text').at(12)).toHaveText('Fazer login com o Google');
    expect(wrapper.find('Text').at(14)).toHaveStyle(styles.txtBtnOthersLogin);
    expect(wrapper.find('Text').at(14)).toHaveText('Cadastro');
    expect(wrapper.find('Text').at(16)).toHaveStyle(styles.txtBtnOthersLogin);
    expect(wrapper.find('Text').at(16)).toHaveText('Esqueci a senha');
  });
  it('renders TouchableOpacity correctly', () => {
    expect(wrapper.find('TouchableOpacity').at(0)).toHaveStyle(styles.midBtn);
    expect(wrapper.find('TouchableOpacity').at(0)).toHaveProp('onPress');
    expect(wrapper.find('TouchableOpacity').at(1)).toHaveStyle(
      styles.btnFbGoogleLogin
    );
    expect(wrapper.find('TouchableOpacity').at(2)).toHaveStyle(
      styles.btnFbGoogleLogin
    );
    expect(wrapper.find('TouchableOpacity').at(3)).toHaveProp('onPress');
    expect(wrapper.find('TouchableOpacity').at(4)).toHaveProp('onPress');
  });
  it('renders Icon correctly', () => {
    expect(wrapper.find('Icon').at(0)).toHaveProp('name', 'arrowright');
    expect(wrapper.find('Icon').at(0)).toHaveProp('size', 35);
    expect(wrapper.find('Icon').at(0)).toHaveProp('color', gray.iron());
    expect(wrapper.find('Icon').at(0)).toHaveProp('style', styles.arrow);
    expect(wrapper.find('Icon').at(1)).toHaveProp('name', 'facebook');
    expect(wrapper.find('Icon').at(1)).toHaveProp('size', 24);
    expect(wrapper.find('Icon').at(1)).toHaveProp('color', purple.frenchLilacLighter());
    expect(wrapper.find('Icon').at(1)).toHaveProp(
      'style',
      styles.iconBtnFbGoogle
    );
    expect(wrapper.find('Icon').at(2)).toHaveProp('name', 'google');
    expect(wrapper.find('Icon').at(2)).toHaveProp('size', 24);
    expect(wrapper.find('Icon').at(2)).toHaveProp('color', purple.frenchLilacLighter());
    expect(wrapper.find('Icon').at(2)).toHaveProp(
      'style',
      styles.iconBtnFbGoogle
    );
  });
});
