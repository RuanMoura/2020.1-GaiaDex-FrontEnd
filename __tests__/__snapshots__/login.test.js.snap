// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders everything 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#f2d9e6",
      "flex": 0,
      "height": 1334,
      "justifyContent": "flex-end",
      "paddingLeft": "8.5%",
      "paddingRight": "8.5%",
      "paddingTop": 0,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#19BB53",
        "borderRadius": 432,
        "height": 432,
        "left": 226,
        "position": "absolute",
        "top": -35,
        "width": 432,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#541E5C",
        "borderRadius": 723,
        "height": 723,
        "left": -177,
        "position": "absolute",
        "top": -416,
        "width": 723,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#0582CA",
        "borderRadius": 432.49,
        "height": 432.49,
        "left": -156.65,
        "position": "absolute",
        "top": -300.62,
        "width": 432.49,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#F2E0F5",
        "flex": 1,
        "fontSize": 60,
        "fontStyle": "normal",
        "fontWeight": "600",
        "lineHeight": 60,
        "marginTop": 184,
      }
    }
  >
    Seja bem-vindo
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Seu e-mail:"
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#e4b4cd",
        "borderColor": "#541E5C",
        "borderRadius": 20,
        "borderWidth": 0.5,
        "color": "black",
        "fontSize": 15,
        "fontStyle": "normal",
        "fontWeight": "500",
        "height": "8%",
        "marginBottom": 20,
        "paddingLeft": 30,
        "paddingRight": 30,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Sua senha:"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#e4b4cd",
        "borderColor": "#541E5C",
        "borderRadius": 20,
        "borderWidth": 0.5,
        "color": "black",
        "fontSize": 15,
        "fontStyle": "normal",
        "fontWeight": "500",
        "height": "8%",
        "marginBottom": 20,
        "paddingLeft": 30,
        "paddingRight": 30,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#242528",
          "fontSize": 40,
          "fontStyle": "normal",
          "fontWeight": "600",
          "lineHeight": 40,
          "marginTop": 12.75,
        }
      }
    >
      Login
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-end",
          "backgroundColor": "#4D525A",
          "borderRadius": 100,
          "height": 8.5,
          "justifyContent": "center",
          "marginBottom": 20,
          "opacity": 1,
          "padding": 35,
          "width": 8.5,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#45194d",
        "borderColor": "#E15A97",
        "borderRadius": 10,
        "height": "6.20%",
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "#F2E0F5",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "500",
            "letterSpacing": 0.1,
            "textAlign": "center",
          }
        }
      >
        Fazer login com o Facebook
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#45194d",
        "borderColor": "#E15A97",
        "borderRadius": 10,
        "height": "6.20%",
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "#F2E0F5",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "500",
            "letterSpacing": 0.1,
            "textAlign": "center",
          }
        }
      >
        Fazer login com o Google
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#242528",
            "fontSize": 18,
            "fontStyle": "normal",
            "fontWeight": "300",
            "letterSpacing": 0.1,
            "marginBottom": 40,
            "textAlign": "center",
          }
        }
      >
        Cadastro
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#242528",
            "fontSize": 18,
            "fontStyle": "normal",
            "fontWeight": "300",
            "letterSpacing": 0.1,
            "marginBottom": 40,
            "textAlign": "center",
          }
        }
      >
        Esqueci a senha
      </Text>
    </View>
  </View>
</View>
`;
